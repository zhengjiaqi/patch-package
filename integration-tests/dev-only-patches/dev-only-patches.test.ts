import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "dev-only-patches",
  shouldProduceSnapshots: true,
})
