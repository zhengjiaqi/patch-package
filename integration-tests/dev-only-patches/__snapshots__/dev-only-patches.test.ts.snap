// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test dev-only-patches: fake-package should be skipped 1`] = `
"SNAPSHOT: fake-package should be skipped
patch-package 0.0.0
Applying patches...
Skipping dev-only fake-package@3.0.0 ✔
left-pad@1.3.0 ✔
Skipping dev-only slash@3.0.0 ✔
END SNAPSHOT"
`;

exports[`Test dev-only-patches: patch-package fails to find fake-package 1`] = `
"SNAPSHOT: patch-package fails to find fake-package
Error: Patch file found for package fake-package which is not present at node_modules/fake-package

  If this package is a dev dependency, rename the patch file to
  
    fake-package+3.0.0.dev.patch

---
patch-package finished with 1 error(s).
END SNAPSHOT"
`;

exports[`Test dev-only-patches: patch-package happily ignores slash because it's a dev dep 1`] = `
"SNAPSHOT: patch-package happily ignores slash because it's a dev dep
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 ✔
Skipping dev-only slash@3.0.0 ✔
END SNAPSHOT"
`;
